### PR TITLE
Docker Compose & Context Middleware Added

### PR DESCRIPTION
**Pull request by iraycd**

### CODE REVIEW

File: src/middleware/get-context.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { asValue } from 'awilix'
+
+export const getContext = async (ctx, next) => {
+  ctx.state.container.register({
+    userContext: asValue({
+      user: 'username'
+    })
+  })
+  next()
```
Comment (jeffijoe) [inline src/middleware/get-context.js]: Needs to be either awaited or returned.

**Comment by iraycd** [comment]
@jeffijoe It is raising issues at 

https://github.com/jeffijoe/koa-es7-boilerplate/blob/6410c1b12ef91488ff3825c4f714f8786f5bc7e3/src/__tests__/todos-api.test.js#L44

While I am adding a context. The test is running successfully, I guess the error is at `fejl`

File: src/middleware/get-context.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { asValue } from 'awilix'
+
+export const getContext = async (ctx, next) => {
```
Comment (jeffijoe) [inline src/middleware/get-context.js]: Maybe add a comment here explaining the purpose.

File: docker-compose.test.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+version: '3.6'
```
Comment (jeffijoe) [inline docker-compose.test.yml:1]: Don't think this file is needed.

File: src/lib/server.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { logger } from './logger'
 import { configureContainer } from './container'
 import { notFoundHandler } from '../middleware/not-found'
 import { errorHandler } from '../middleware/error-handler'
+import { getContext } from '../middleware/get-context'
```
Comment (jeffijoe) [inline src/lib/server.js]: Let's call this `registerContext` (file: `register-context`)

**Review by jeffijoe — CHANGES_REQUESTED** [review]
Hey Ray, thanks for the PR! There's a few things that needs changing before I can merge this.

**Comment by jeffijoe** [comment]
@iraycd the issue is cause of the missing `await`/`return` as I mentioned in the review. It's causing the request to end early.

File: docker-compose.test.yml (hunk 3 above)
Comment (iraycd) [reply docker-compose.test.yml:1]: I have added this to make life of testing easy while using database.

**Comment by iraycd** [comment]
@jeffijoe Got it. Sorry

File: docker-compose.test.yml (hunk 3 above)
Comment (jeffijoe) [reply docker-compose.test.yml:1]: Ah, I usually just use a single docker compose for development. What does `hms` mean in this context?

File: src/middleware/register-context.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { asValue } from 'awilix'
+
+// Register Context helps to add request-specific data to the scope.
+export const registerContext = async (ctx, next) => {
```
Comment (jeffijoe) [inline src/middleware/register-context.js:4]: Oh, just noticed the other middleware is declared as a regular async function, and also uses docblock style comments, 

```
/**
 * Like this
 */
```

**Review by jeffijoe — CHANGES_REQUESTED** [review]
Sorry! Noticed one more thing 😅

**Comment by jeffijoe** [comment]
Also, if you insist on adding the docker compose files, lets rename `hms` to `app` — `hms` must have been a left-over from your own project I assume?

File: docker-compose.test.yml (hunk 3 above)
Comment (iraycd) [reply docker-compose.test.yml:1]: renamed to `app` and added `readme.md`

**Review by jeffijoe — APPROVED** [review]
Looks good!

**Comment by jeffijoe** [comment]
Squashed and merged, thanks @iraycd !

File: src/middleware/register-context.js (hunk 5 above)
Comment (iraycd) [reply src/middleware/register-context.js:4]: Should I change the other middleware?

Comment (jeffijoe) [reply src/middleware/register-context.js:4]: Oh snap, I was too quick to merge! Preferably change this one to use function, arrow funcs are usually used inside functions, while functions are the top-level (they have no `this` binding anyway)

**Pull request merged**